### PR TITLE
feat: add method to check pvc exists

### PR DESCRIPTION
**Pull request by inshbha2**
feat: add method to check pvc exists with given status in the given namespace

### CODE REVIEW

File: pkg/kube/structured/structured.go [hunk 1]
```diff
@@ -235,6 +235,18 @@ func PersistentVolExists(kubeClientset kubernetes.Interface, name, expectedPhase
...
+	if err != nil {
+		return err
+	}
+	phase := string(vol.Status.Phase)
+	if phase != expectedPhase {
+		return fmt.Errorf("persistentvolumeclaim had unexpected phase %v, expected phase %v", phase, expectedPhase)
+	}
+	return nil
+}
+
```
Comment (garomonegro) [inline pkg/kube/structured/structured.go:249]: Is it possible for the claim to exists but have not yet converge to the desire phase? Meaning the claim is now in another phase but will eventually get to the desired phase. If so, retrying here would be worth it. WDYT?

File: pkg/kube/structured/structured_test.go [hunk 2]
```diff
@@ -480,6 +481,50 @@ func TestPersistentVolExists(t *testing.T) {
...
+		{
+			name: "pvc not found Test",
+			args: args{
+				kubeClientset: fake.NewSimpleClientset(getResource(t, persistentVolumeClaimType, "testabc")),
+				name:          persistentvolumeClaimName,
+				namespace:     "",
+				expectedPhase: "",
+			},
+			wantErr: true,
+		},
```
Comment (garomonegro) [inline pkg/kube/structured/structured_test.go:517]: Could you add an expectedPhase to the first case, as well as a test case where the expectedPhase is not the actual phase?

**Comment by garomonegro** [comment]
Could you run make generate so that the syntax documentation gets updated?

File: pkg/kube/structured/structured.go (hunk 1 above)
Comment (inshbha2) [reply pkg/kube/structured/structured.go:249]: yes, will add retry

File: pkg/kube/structured/structured_test.go (hunk 2 above)
Comment (inshbha2) [reply pkg/kube/structured/structured_test.go]: done

File: pkg/kube/structured/structured.go [hunk 3]
```diff
@@ -235,6 +235,20 @@ func PersistentVolExists(kubeClientset kubernetes.Interface, name, expectedPhase
...
+	})
+	if err != nil {
+		return err
+	}
+	phase := string(vol.(*corev1.PersistentVolumeClaim).Status.Phase)
+	if phase != expectedPhase {
+		return fmt.Errorf("persistentvolumeclaim had unexpected phase %v, expected phase %v", phase, expectedPhase)
+	}
+	return nil
+}
```
Comment (garomonegro) [inline pkg/kube/structured/structured.go:250]: You are only retrying the call to GetPersistentVolumeClaim which I already has retry, you need to retry the whole thing:
``` go
PersistentVolClaimExists(){
  RetryOnError(){
    GetPersistentVolumeClaim(){
      RetryOnError(){ // already there
        GetPVC(){}
      }
    }
    CheckPhase(){}
  }
}
```

Comment (inshbha2) [reply pkg/kube/structured/structured.go]: yes, done.. thanks

**Review by garomonegro — APPROVED** [review]

**Pull request merged**